### DISCUSSION TITLE
How will I update to the latest version?

### DISCUSSION THREAD

**Discussion by rafsko1** [immich-app/immich#1288] [General]
How will I update to the latest version on RPI with Debian Bullseye Lite 64bit? Is watchtower safe?

**Comment by alextran1502** [comment]
Please refer to this guide here https://immich.app/docs/install/docker-compose/#upgrading

**Reply by smileBeda** [reply]
New link: https://immich.app/docs/install/docker-compose/#upgrading

**Reply by NicholasFlamy** [reply]
> New link: https://immich.app/docs/install/docker-compose/#upgrading

Updated the comment.

**Comment by bo0tzz** [comment]
> Is watchtower safe?

I would recommend against watchtower - Immich is under heavy development, and sometimes you may need to take some action while upgrading.  So instead, you should read the release notes and then perform an upgrade manually.

**Comment by rafsko1** [comment]
Thanks a mill.

**Comment by rafsko1** [comment]
Getting error like this: 

permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json?all=1&filters=%7B%22label%22%3A%7B%22com.docker.compose.project%3Dpi%22%3Atrue%7D%7D": dial unix /var/run/docker.sock: connect: permission denied



After running this:

sudo docker compose pull && docker compose up -d

**Reply by bo0tzz** [reply]
You need to use sudo for the up command as well

**Comment by delirious11** [comment]
what's the procedure for portainer stack?

**Reply by DavidTIBO** [reply]
Same question here !

**Reply by dudaskank** [reply]
First, check the breaking changes (https://github.com/immich-app/immich/discussions?discussions_q=sort%3Adate_created+label%3Abreaking-change) if you have to do some manual changes in variables or whatever.

After checking that everything is ok, in Portainer, stacks, enter the Immich stack, Editor panel, scroll down after the variables and click "Update the stack" button in Actions section, check "Re-pull image and redeploy" and click "Update".

**Comment by NicholasFlamy** [comment]
I think this can be closed.
